### PR TITLE
Add documentation on partial segment matching

### PR DESCRIPTION
**Pull request by miss85246**
modify `routing.md` file to add segment matching part

### CODE REVIEW

**Comment by miss85246** [comment]
image

File: src/en/guide/basics/routing.md [hunk 1]
```diff
@@ -277,7 +280,43 @@ async def handler(request, foo: str):
...
+
+But for complex routing, the above example is still too simple.
+
+Sometimes you want to match on a part of a path segment:
+
+```text
+/image/123456789.jpg
+```
+
+If you wanted to match the file pattern, but only capture the numeric portion, you need to do some regex fun:
```
Comment (ahopkins) [inline src/en/guide/basics/routing.md:294]: Feels like we needs some emoji here to make it more "fun" :smile: Who doesn't like seeing more emojis :flags:

File: src/en/guide/basics/routing.md [hunk 2]
```diff
@@ -277,7 +280,43 @@ async def handler(request, foo: str):
...
+
+```python
+app.route(r"/image/<img_id:(?P<img_id>\d+)\.jpg>")
```
Comment (ahopkins) [inline src/en/guide/basics/routing.md:297]: I think we should offer this as the standard pattern and not scare anyone away with named matching if we do not need to:

```python
app.get(r"/<foo:(\d+).jpg>")
```

File: src/en/guide/basics/routing.md [hunk 3]
```diff
@@ -277,7 +280,43 @@ async def handler(request, foo: str):
...
+```
+
+Further, these should all be acceptable:
+
+```python
+@app.get(r"/<foo:[a-z]{3}.txt>")                # matching on the full pattern
+@app.get(r"/<foo:([a-z]{3}).txt>")              # defining a single matching group
+@app.get(r"/<foo:(?P<foo>[a-z]{3}).txt>")       # defining a single named matching group
+@app.get(r"/<foo:(?P<foo>[a-z]{3}).(?:txt)>")   # defining a single named matching group, with one or more non-matching groups
```
Comment (ahopkins) [inline src/en/guide/basics/routing.md:306]: I think we should make all of these examples use the `\d+` with `jpg` pattern for consistency with the first example we give.

Maybe the last one can be an exception and be a little different:

```python
r"/<foo:(?P<foo>\d+).(?:jpg|png)>
```

File: src/en/guide/basics/routing.md [hunk 4]
```diff
@@ -277,7 +280,43 @@ async def handler(request, foo: str):
...
+```
+
```
Comment (ahopkins) [inline src/en/guide/basics/routing.md:308]: > Therefore, we can see that when using regex routing, we have 4 options:

File: src/en/guide/basics/routing.md [hunk 5]
```diff
@@ -277,7 +280,43 @@ async def handler(request, foo: str):
...
+1. matching on the full pattern
+2. defining a single matching group
+3. defining a single named matching group
+4. defining a single named matching group, with one or more non-matching groups
+
+Also, if using a named matching group, it must be the same as the segment label.
```
Comment (ahopkins) [inline src/en/guide/basics/routing.md:314]:
```
app.get(r"/<foo:(?P<foo>\d+).jpg>")  # OK
app.get(r"/<foo:(?P<bar>\d+).jpg>")  # NOT OK
```

**Review by ahopkins — CHANGES_REQUESTED** [review]
This is super helpful. I am somewhat reluctant to hide it away in the regex tab.

What do you think about adding a new subsection after the tabs for "Partial segment matching"? And instead having a mention in the regex tab to "see below" for more details?

**Comment by miss85246** [comment]
good idea, I will add a new subsection in my free time 😄

**Pull request closed without merging**